core/src/main/java/com/oath/oak/ValueUtilsImpl.java

```diff
@@ -165,15 +172,26 @@ public ValueResult compute(Slice s, Consumer<OakWBuffer> computer, int version)
         if (valueDeserializeTransformer != null) {
             oldValue = valueDeserializeTransformer.apply(getValueByteBufferNoHeader(lookUp.valueSlice));
         }
-        Slice s = innerPut(chunk, lookUp, value, serializer, memoryManager);
+        Slice s = lookUp.valueSlice;
+        int capacity = serializer.calculateSize(value);
+        if (capacity + getHeaderSize() > s.getByteBuffer().remaining()) {
+            s = moveValue(chunk, lookUp, memoryManager, internalOakMap, value);
+            if (s == null) {
+                // rebalance was needed or the entry was updated by someone else, need to retry
+                unlockWrite(lookUp.valueSlice);
+                return Result.withFlag(RETRY);
+            }
+        } else {
+            s = innerPut(chunk, lookUp, value, serializer, memoryManager);
+        }
         unlockWrite(s);
```

Comment:
I do not see how to release lock on the old and new slice....

Reply:
You don't need to release the lock of the new slice since you never locked it.
But the lock of the old slice must be unlocked.
The current code unlocks the slice object `s` because it assumes it refers to the same object that is in `lookUp`: `Slice s = lookUp.valueSlice;`.
It is very confusing. Why not lock the exact same object explicitly, i.e., `unlockWrite(lookUp.valueSlice);`?